symfit/core/fit.py

```diff
@@ -1235,93 +1027,147 @@ def execute(self, relative_error=1e-8, max_iter=500):
 
+        # Use Nelder-Mead instead
+        fit = Fit(model, x=xdata, y=ydata, minimizer=NelderMead)
+        fit_result = fit.execute()
 
-class Minimize(BaseFit):
     """
-    Minimize a model subject to constraints. A wrapper for ``scipy.optimize.minimize``.
 
-    ``Minimize`` itself doesn't work when data is provided to its Variables, use
-    one of its subclasses for that.
-    """
-    @keywordonly(constraints=None)
-    def __init__(self, model, *args, **kwargs):
+    @keywordonly(objective=None, minimizer=None, constraints=None)
+    def __init__(self, model, *ordered_data, **named_data):
         """
-        Because in a lot of use cases for Minimize no data is supplied to variables,
-        all the empty variables are replaced by an empty np array.
 
-        :constraints: constraints the minimization is subject to.
-        :type constraints: list
+        :param model: (dict of) sympy expression(s) or ``Model`` object.
+        :param constraints: iterable of ``Relation`` objects to be used as
+            constraints.
+        :param bool absolute_sigma: True by default. If the sigma is only used
+            for relative weights in your problem, you could consider setting it to
+            False, but if your sigma are measurement errors, keep it at True.
+            Note that curve_fit has this set to False by default, which is wrong in
+            experimental science.
+        :param objective: Have Fit use your specified objective. Can be one of
+            the predifined `symfit` objectives or any callable which accepts fit
+            parameters and returns a scalar.
+        :param minimizer: Have Fit use your specified minimizer.
+        :param ordered_data: data for dependent, independent and sigma variables. Assigned in
+            the following order: independent vars are assigned first, then dependent
+            vars, then sigma's in dependent vars. Within each group they are assigned in
+            alphabetical order.
+        :param named_data: assign dependent, independent and sigma variables data by name.
         """
-        constraints = kwargs.pop('constraints')
-        super(Minimize, self).__init__(model, *args, **kwargs)
+        objective = named_data.pop('objective')
+        minimizer = named_data.pop('minimizer')
+        constraints = named_data.pop('constraints')
+        super(Fit, self).__init__(model, *ordered_data, **named_data)
+
+        # List of Constraint objects
+        self.constraints = self._init_constraints(constraints=constraints)
+
+        if objective is None:
+            # Param only scalar Model -> the model is the objective.
+            if len(self.model.independent_vars) == 0 and len(self.model) == 1:
+                # No data provided means a simple minimization of the Model parameters
+                # is requested, not a fit.
+                if all(value is None for value in self.data.values()):
+                    objective = MinimizeModel
```

Comment:
I agree that currently it could be `and`-ed , but since both are already very long statements I don't think it will help readability.

An nice example of something that doesn't match the second if btw would be `y = a`, which means that `a` will be the mean value of `y`.  Nice eh?